data_steward/bq_utils.py

```diff
@@ -739,6 +739,31 @@ def create_dataset(
     return insert_result
 
 
+def get_sql_row_expr(row: dict, fields: list):
+    """
+    Get appropriate SQL expression for a row based on fields spec
+
+    :param row: csv row with untyped values
+    :param fields: bigquery fields spec
+    :return: format string for row object
+    """
+    val_exprs = []
+    # TODO refactor for all other types or use external library
+    for field_name, val in row.items():
+        field = next(filter(lambda f: f['name'] == field_name, fields), None)
+        if field is None:
+            raise InvalidOperationError(f'Unable to marshal {val}: field "{field_name}" was not found')
+        if not val and field['mode'] == 'nullable':
```

Comment:
Can `val` be a numeric type here? If so you would be inserting NULL instead of 0 for nullable int fields which may not be desired

Either way, I would add a test case for this

Reply:
I think `val` shouldn't be numeric here. I will add a test case.